contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/ApexKuduConnection.java

```diff
@@ -0,0 +1,223 @@
+
+  private transient  KuduSession kuduSession;
+
+  private transient  KuduTable kuduTable;
+
+  private transient  KuduClient kuduClient;
+
+  Logger LOG = LoggerFactory.getLogger(ApexKuduConnection.class);
+
+
+  private ApexKuduConnection(ApexKuduConnectionBuilder builder)
+  {
+    checkNotNull(builder,"Builder cannot be null to establish kudu session");
+    checkArgument(builder.mastersCollection.size() > 0, "Atleast one kudu master needs to be specified");
+    checkNotNull(builder.tableName,"Kudu table cannot be null");
+    KuduClient.KuduClientBuilder kuduClientBuilder = new KuduClient.KuduClientBuilder(builder.mastersCollection);
+    if (builder.isOperationTimeOutSet) {
+      kuduClientBuilder.defaultOperationTimeoutMs(builder.operationTimeOutMs);
+    }
+    if (builder.isBossThreadCountSet) {
+      kuduClientBuilder.bossCount(builder.numBossThreads);
+    }
+    if (builder.isWorkerThreadsCountSet) {
+      kuduClientBuilder.workerCount(builder.workerThreads);
+    }
+    if (builder.isSocketReadTimeOutSet) {
+      kuduClientBuilder.defaultSocketReadTimeoutMs(builder.socketReadTimeOutMs);
+    }
+    kuduClient = kuduClientBuilder.build();
+    kuduSession = kuduClient.newSession();
+    if (builder.isFlushModeSet) {
+      kuduSession.setFlushMode(builder.flushMode);
+    }
+    if (builder.isExternalConsistencyModeSet) {
+      kuduSession.setExternalConsistencyMode(builder.externalConsistencyMode);
+    }
+    try {
+      if (!kuduClient.tableExists(builder.tableName)) {
+        throw new Exception("Table " + builder.tableName + " does not exist. ");
+      } else {
+        kuduTable = kuduClient.openTable(builder.tableName);
+      }
+    } catch (Exception e) {
+      LOG.error("Kudu table existence could not be ascertained  " + e.getMessage());
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void close() throws Exception
+  {
+    kuduSession.close();
+    kuduClient.close();
+  }
+
+  public KuduSession getKuduSession()
+  {
+    return kuduSession;
+  }
+
+  public void setKuduSession(KuduSession kuduSession)
+  {
+    this.kuduSession = kuduSession;
+  }
+
+  public KuduTable getKuduTable()
+  {
+    return kuduTable;
+  }
+
+  public void setKuduTable(KuduTable kuduTable)
+  {
+    this.kuduTable = kuduTable;
+  }
+
+  public KuduClient getKuduClient()
+  {
+    return kuduClient;
+  }
+
+  public void setKuduClient(KuduClient kuduClient)
+  {
+    this.kuduClient = kuduClient;
+  }
+
+  public static class ApexKuduConnectionBuilder
+  {
+    List<String> mastersCollection = new ArrayList<>();
+
+    String tableName;
+
+    // optional props
+    int numBossThreads = 1;
+
+    boolean isBossThreadCountSet = false;
+
+    int workerThreads = 2 *  Runtime.getRuntime().availableProcessors();
+
+    boolean isWorkerThreadsCountSet = false;
+
+    long socketReadTimeOutMs = 10000;
+
+    boolean isSocketReadTimeOutSet = false;
+
+    long operationTimeOutMs = 30000;
```

Comment:
Well this is part of the builder pattern and hence:
1. It cannot be final
2. It is ideal this not be static as well as there might be use cases where one would like to use multiple instances of the kudu output operator in a single application each one having its own connection properties that would be built by the builder. 
3. It is not public because it is part of the builder pattern.